rust/tests/datafusion_test.rs

```diff
@@ -228,8 +228,31 @@ async fn test_datafusion_stats() -> Result<()> {
     Ok(())
 }
 
+async fn get_scan_metrics(
```

Comment:
not strictly required in this PR, but we have a very basic implementation for a metrics collector, that I plan to promote to a first class delta-rs member soon. The aim would be to harmonise our approach to metrics collection across operations. Just in case you also want to use that here :)

https://github.com/delta-io/delta-rs/blob/8dcf46e614370b93a6d47402456621b713768674/rust/tests/datafusion_test.rs#L39-L59

Reply:
Yeah I'm already using this implementation with a minor change if there are zero records. I found it really useful!